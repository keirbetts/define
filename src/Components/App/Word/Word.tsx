import { useState, useEffect } from "react";
import { getWord } from "../../../services/getWordService/getWordService";
import { randomWordGenerator } from "../../../utils/words";
import { TWENTY_FOUR_HOURS } from "../../../constants/intervals";



export const Word = () => {

  const [activeWord, setActiveWord] = useState<string>("");

  const getWordForGuess = async () => {

    let newWord = randomWordGenerator();
    const response = await getWord(newWord);
    setActiveWord(response);
  };

  useEffect(() => {
    const interval = setInterval(() => {
      getWordForGuess();
      console.log("Logs every 24 hours");
    }, 6000);

    return () => clearInterval(interval);
  }, []);


  console.log(activeWord, "word");

  // TODO: user guess and submit - remove weird symbols returned from api

  return (
    <div className="word-wrapper">
      Define
      < br />
      <br />
      {activeWord}
      <br />
      <br />
      <textarea
        placeholder="Enter word definition..."
      ></textarea>
      <br />
      <br />
      <button>
        Submit guess
      </button>
    </div>
  );
};