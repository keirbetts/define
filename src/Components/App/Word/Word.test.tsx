import { render } from "@testing-library/react";
import { Word } from "./Word";



describe("<Word />", () => {

  it("renders successfully", async () => {
    const { container } = render(<Word />);

    expect(container.firstChild).toMatchSnapshot();
  });
});