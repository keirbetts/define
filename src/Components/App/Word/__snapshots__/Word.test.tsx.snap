// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Word /> renders successfully 1`] = `
<div
  class="word-wrapper"
>
  Define
  <br />
  <br />
  <br />
  <br />
  <textarea
    placeholder="Enter word definition..."
  />
  <br />
  <br />
  <button>
    Submit guess
  </button>
</div>
`;
